fhir-persistence-params/src/main/java/com/ibm/fhir/persistence/params/database/PlainParamValueProcessor.java

```diff
@@ -0,0 +1,834 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.persistence.params.database;
+
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.params.api.BatchParameterValue;
+import com.ibm.fhir.persistence.params.api.IParamValueProcessor;
+import com.ibm.fhir.persistence.params.api.IParameterIdentityCache;
+import com.ibm.fhir.persistence.params.api.ParamSchemaConstants;
+import com.ibm.fhir.persistence.params.model.CodeSystemValue;
+import com.ibm.fhir.persistence.params.model.CommonCanonicalValue;
+import com.ibm.fhir.persistence.params.model.CommonCanonicalValueKey;
+import com.ibm.fhir.persistence.params.model.CommonTokenValue;
+import com.ibm.fhir.persistence.params.model.CommonTokenValueKey;
+import com.ibm.fhir.persistence.params.model.LogicalResourceIdentKey;
+import com.ibm.fhir.persistence.params.model.LogicalResourceIdentValue;
+import com.ibm.fhir.persistence.params.model.ParameterNameValue;
+
+/**
+ * Loads search parameter values into the target PostgreSQL database using
+ * the plain (non-sharded) schema variant.
+ */
+public abstract class PlainParamValueProcessor implements IParamValueProcessor {
+    private static final String CLASSNAME = PlainParamValueProcessor.class.getName();
+    private static final Logger logger = Logger.getLogger(PlainParamValueProcessor.class.getName());
+    protected static final short FIXED_SHARD = 0;
+
+    // the connection to use for the inserts
+    protected final Connection connection;
+
+    // The translator to handle variations in SQL syntax
+    protected final IDatabaseTranslator translator;
+
+    // The FHIR data schema
+    protected final String schemaName;
+
+    // the cache we use for various lookups
+    protected final IParameterIdentityCache identityCache;
+
+    // The processor used to process the batched parameter values after all the reference values are created
+    private final PlainBatchParameterProcessor batchProcessor;
```

Comment:
Here is the relationship between IParamValueProcessor implementations and IBatchParameterProcessors that I was looking for.  Our PlainParamValueProcessor *uses* a PlainBatchParameterProcessor to do its job of inserting parameter values to the DB.